api: Add `params_show` (#3459)
iterative/dvc.org: content/docs/api-reference/params_show.md

# dvc.api.params_show()

Load <abbr>parameters</abbr> (name and values) tracked in a <abbr>DVC
project</abbr>.

```py
def params_show(
    *targets: str,  # Optional
    stages: Optional[Union[str, Iterable[str]]] = None,
    repo: Optional[str] = None,
    rev: Optional[str] = None,
    deps: bool = False,
) -> Dict:
```

## Usage:

```py
import dvc.api

params = dvc.api.params_show()
```

## Description

Retrieves <abbr>params</abbr> keys and values from a <abbr>DVC project</abbr>
and returns a dictionary, such as:

```json
{
  "split": 0.2,
  "seed": 20170428
}
```

Without arguments, this function will retrieve all params from all tracked
parameter files (used in any `dvc.yaml` file). This applies to the current
project version when using Git (including any changes in the working tree).

The function parameters (below) let you restrict what's retrieved.

## Parameters

- `*targets` - paths to one or more valid parameter file(s) to retrieve params

>   from. For example, `"params.py, myparams.toml"`. If no `targets` are provided,

Reviewer: ```suggestion
  from. For example, `"params.py", "myparams.toml"`. If no `targets` are provided,
```

TBH I would suggest removing that sentence. I don't see similar examples for targets in the CLI, and I think people should know how Python positional args work?

Reviewer 2: PTAL

  all param files tracked in any `dvc.yaml` will be used by default. Note that
  explicit targets don't have to be used in a `dvc.yaml` (unless `deps=True`).

- `repo` - specifies the location of the DVC project. It can be a URL or a file
  system path. Both HTTP and SSH protocols are supported for online Git repos
  (e.g. `[user@]server:project.git`). _Default_: The current project (found by
  walking up from the current working directory tree).

- `stages` - one or more names of the stage(s) to retrieve params from.
  _Default_: `None` (all parameters from all stages will be retrieved).

- `rev` - Git commit (any [revision](https://git-scm.com/docs/revisions) such as
  a branch or tag name, a commit hash or an
  [experiment](/doc/command-reference/exp) name). If `repo` is not a Git repo,
  this option is ignored. _Default_: `None` (current working tree will be used)

- `deps` - whether to retrieve only params that are stage dependencies. Accepts
  `True` or `False` (_default_).

## Example: Filter by stage name(s)

> Working on https://github.com/iterative/example-get-started, file

> > `src/featurization.py`.

Reviewer: I think we are better off without referencing an example repo because:

1. Users probably aren't familiar with the repo like we are and shouldn't have to check out the repo to understand the context.
2. We probably should update the example repo with `dvc.api.params_show()`, and then we will have to update this ref again.

Reviewer: To clarify, all the example content underneath looks good. I just don't think we should reference that it's based on example-get-started.

Reviewer 2: > all the example content underneath looks good

But rn the example uses a git-diff style so it's supposed to be a change based on something:

![image](https://user-images.githubusercontent.com/1477535/176541503-8ceeb00b-3f53-450e-8c46-4d04b744de09.png)

@daavoo mind rewriting this part so no reference is needed yet the usage of the function is explained in the code snippet? Prob no need for git-diff block after all.

`stages` can be a single name (string):

```git
+import json
-import yaml

-params = yaml.safe_load(open("params.yaml"))["featurize"]
+import dvc.api
+

> +params = dvc.api.get_params(stages="featurize")

Author: ```suggestion
+params = dvc.api.params_show(stages="featurize")
```

+params = json.dumps(params, indent=2)

 ...
-max_features = params["max_features"]
-ngrams = params["ngrams"]
+max_features = params["featurize"]["max_features"]
+ngrams = params["featurize"]["ngrams"]
```

Or an iterable of strings:

```py
import json
import dvc.api

> params = dvc.api.get_params(stages=["featurize", "train"])

Author: ```suggestion
params = dvc.api.params_show(stages=["featurize", "train"])
```

```

```json
{
  "featurize": {
    "max_features": 200,
    "ngrams": 2
  },
  "train": {
    "seed": 20170428,
    "n_est": 50,
    "min_split": 0.01
  }
}
```

## Example: Load specific parameter file(s)

You can pass any valid param file path as target to load all of the parameters
defined in it:

```py
import json
import dvc.api
params = dvc.api.params_show("params.yaml")
```

```json
{
  "run_mode": "prod",
  "configs": {
    "dev": "configs/params_dev.yaml",
    ...
```

Or multiple path targets:

```py
import json
import dvc.api
params = dvc.api.params_show(
  "configs/params_dev.yaml", "configs/params_prod.yaml")
```

```json
{
  "configs/params_prod.yaml:run_mode": "prod",
  "configs/params_prod.yaml:config_file": "configs/params_prod.yaml",
  ...
  "configs/params_dev.yaml:run_mode": "dev",
  "configs/params_dev.yaml:config_file": "configs/params_dev.yaml",
  ...
```

## Example: Use a remote DVC repository

You can use the `repo` argument to retrieve parameters from any <abbr>DVC
repository</abbr> without having to clone it locally.

```py
import json
import dvc.api
params = dvc.api.get_params(

>     repo="https://github.com/iterative/demo-fashion-mnist")

Reviewer: Can we use example-get-started here since it's stable and already a docs dependency?

Reviewer 2: Agree.

```

```json
{
  "train": {
    "batch_size": 128,
    "hidden_units": 64,
    "dropout": 0.4,
    ...
```

## Example: Specify a project version

> Working on https://github.com/iterative/example-get-started

You can retrieve params from arbitrary Git commits, for example a branch name:

```py
import json
import dvc.api

> params = dvc.api.get_params(rev="tune-hyperparams")

Author: ```suggestion
params = dvc.api.params_show(rev="tune-hyperparams")
```

# dvc.api.params_show()

Load <abbr>parameters</abbr> (name and values) tracked in a <abbr>DVC
project</abbr>.

```py
def params_show(
    *targets: str,  # Optional
    repo: Optional[str] = None,

>     stages: Optional[Union[str, Iterable[str]]] = None,

Reviewer 2: This should be the real function signature. Did it change?

>     stages: Optional[Union[str, Iterable[str]]] = None,

Reviewer 2: This is the signature [from what I can tell](https://github.com/iterative/dvc/pull/7613/files#diff-54fd2c4af8ff96a0ddfb2a22c1c56a659c41ae7eb263a6fb943b6d246cb34025R9-R15). I don't think we should alter it for docs (other than that one commend I added).

>     stages: Optional[Union[str, Iterable[str]]] = None,

Reviewer 2: That said if you plan to change the signature in the core repo then sure. Lmk

>     stages: Optional[Union[str, Iterable[str]]] = None,

Author: Note that those are all keyword arguments so the order doesn't really affect the behavior. We could also update the order in core, if it makes more sense

>     stages: Optional[Union[str, Iterable[str]]] = None,

Reviewer: I don't mind updating in core, but to me it's more important to update here to start with more common use cases.

>     stages: Optional[Union[str, Iterable[str]]] = None,

Reviewer 2: OK. I moved `stages` one position up. Why not all the way up then? Since deem the stage-filtering example most important... Which goes back to https://github.com/iterative/dvc/pull/7613#pullrequestreview-1009761660

> I don't mind updating in core

I think that's ideal because at some point this may get copy-pasted again from there (on some update) and our custom docs order is gone.

    rev: Optional[str] = None,

>     deps: bool = False,

Author: @dberenbaum @jorgeorpinel 

Should we drop `deps`? 

It doesn't feel that useful after adding support for `stages`

Reviewer 2: The only use case I see is to combine param file `targets` and `deps=True` to discover which params in a param file are used by the DVC project. If that's an intended feature we should probably add an example (or change the last one).

Author: > If that's an intended feature we should probably add an example (or change the last one).

I believe it is the original idea behind `deps` in the CLI. 

Note that you don't necessarily need to combine it with `targets`, it could be also useful to discover which params from *any* param file are used by the DVC project

Reviewer 2: If we don't see much use in retrospect, no need for an example. Yet OK to keep to reflect the CLI, IMO. Open an issue in core to deprecate `deps` from both CLI and API maybe?

) -> Dict:
```

## Usage:

```py
import dvc.api

params = dvc.api.params_show()
```

## Description

Retrieves <abbr>params</abbr> keys and values from a <abbr>DVC project</abbr>

> and returns a JSON structure such as:

Reviewer: Isn't it a Python dict and not a JSON structure?

Reviewer 2: My bad. Fixed.

  from. For example, `"params.py, myparams.toml"`. If no `targets` are provided,
  all param files tracked in any `dvc.yaml` will be used by default. Note that
  explicit targets don't have to be used in a `dvc.yaml` (unless `deps=True`).

- `repo` - specifies the location of the DVC project. It can be a URL or a file
  system path. Both HTTP and SSH protocols are supported for online Git repos
  (e.g. `[user@]server:project.git`). _Default_: The current project (found by
  walking up from the current working directory tree).

- `stages` - one or more names of the stage(s) to retrieve params from.
  _Default_: `None` (all parameters from all stages will be retrieved).

- `rev` - Git commit (any [revision](https://git-scm.com/docs/revisions) such as
  a branch or tag name, a commit hash or an
  [experiment](/doc/command-reference/exp) name). If `repo` is not a Git repo,
  this option is ignored. _Default_: `None` (current working tree will be used)

- `deps` - whether to retrieve only params that are stage dependencies. Accepts
  `True` or `False` (_default_).

## Example: Filter by parameter file(s)s

> Working on

> > https://github.com/iterative/pipeline-conifguration/tree/main/multi-params-files

Reviewer 2: Fixed here for now but

a) once renamed, conifguration would redirect to configuration I think (by GH);
b) more importantly: yes, let's definitely not use unofficial example repos in the docs.

> we don't appear to have any proper example
include a minimal example setup

We can keep them al hypothetical here. No need to make cmd ref example sreproducible, it's fine ⌛

`stages` can be a single name (string):

```py
import json
import dvc.api
params = dvc.api.get_params(stages="featurize")
print(json.dumps(params, indent=2))
```

```json
{
  "featurize": {
    "max_features": 200,
    "ngrams": 2
  }
}
```

Or an iterable of strings:

```py
import json
import dvc.api
params = dvc.api.get_params(stages=["featurize", "train"])
print(json.dumps(params, indent=2))
```

```json
{
  "featurize": {
    "max_features": 200,
    "ngrams": 2
  },
  "train": {
    "seed": 20170428,
    "n_est": 50,
    "min_split": 0.01
  }
}
```

## Example: Load specific parameter file(s)

> Working on
> https://github.com/iterative/pipeline-conifguration/tree/main/multi-params-files

You can pass any valid param file path as target to load all of the parameters

> defined in it:

Author: It is accurate. Although I would suggest including a note here clarifying that the param file path doesn't necessarily need to be tracked by DVC

You can retrieve params from arbitrary Git commits, for example a branch name:

```py
import json
import dvc.api
params = dvc.api.get_params(rev="tune-hyperparams")
print(json.dumps(params, indent=2))
```

```json
{
  "prepare": {
    "split": 0.2,
    "seed": 20170428
  },
  "featurize": {
    "max_features": 200,
    "ngrams": 2
  },
  "train": {
    "seed": 20170428,
    "n_est": 100,
    "min_split": 8
  }
}
```

## Example: Use a remote DVC repository

You can use the `repo` argument to retrieve parameters from any <abbr>DVC
repository</abbr> without having to clone it locally.

```py
import json
import dvc.api
params = dvc.api.get_params(
    repo="https://github.com/iterative/demo-fashion-mnist")
print(json.dumps(params, indent=2))
```

```json
{
  "train": {
    "batch_size": 128,
    "hidden_units": 64,
    "dropout": 0.4,
    ...
```

> ## Example: Filter by stage name(s)

Reviewer: Can this be moved up, preferably to the top or at least right below the `targets` example? Also, could you add some context for how this can be used inside a stage, or is that better left for a user guide doc?

Author: > Also, could you add some context for how this can be used inside a stage, or is that better left for a user guide doc?

I included a minimal example on the very first version of the P.R. but it felt too much for API Reference.

Reviewer 2: Sure, we can have an ad-hoc order for examples. Moved all the way up (and ordered the remaining ones to match the parameter order).

> could you add some context for how this can be used inside a stage, or is that better left for a user guide doc?

Who is "you" here? 😬 Let's put it here as we don't have any guides with API usage and probably won't soon(?)

> included a minimal example

Again, no need to make things fully reproducible. I think @dberenbaum means some more sample code that explains what you would do with the params dict -- in this case set the model training method params I suppose?

Reviewer 2: > more sample code that explains what you would do with the params dict

I tried doing this in 2a70b7dcf. PTAL @daavoo @dberenbaum 

In review app: https://dvc-org-api-get-params-2lnv5kl.herokuapp.com/doc/api-reference/params_show#example-filter-by-stage-names

parameter files (used in any `dvc.yaml` file). This applies to the current
project version when using Git (including any changes in the working tree).

The function parameters (below) let you restrict what's retrieved.

## Parameters

- `*targets` - names of one or more valid parameter file to retrieve params
  from. For example, `"params.py, myparams.toml"`. If no `targets` are provided,
  all param files tracked in any `dvc.yaml` will be used by default. Note that
  explicit targets don't have to be used in a `dvc.yaml`, however.

- `repo` - specifies the location of the DVC project. It can be a URL or a file
  system path. Both HTTP and SSH protocols are supported for online Git repos
  (e.g. `[user@]server:project.git`). _Default_: The current project (found by
  walking up from the current working directory tree).

- `stages` - name or names of the stages to retrieve params from. If `None`, all
  parameters from all stages will be retrieved. _Default_: `None`.

- `rev` - Git commit (any [revision](https://git-scm.com/docs/revisions) such as
  a branch or tag name, a commit hash or an
  [experiment](/doc/command-reference/exp) name). If `repo` is not a Git repo,

>   this option is ignored. _Default_: `None` (current working tree will be used)

Reviewer 2: > _Default_: `None` (current working tree will be used)

Is this the same for all the other functions BTW? Currently they state that the default is `HEAD`.

Author: > Is this the same for all the other functions BTW? Currently they state that the default is HEAD.

Yes, it is the same for all the other functions, HEAD can be kind of misleading there

Reviewer 2: Extracted to #3681

- `repo` - specifies the location of the DVC project. It can be a URL or a file
  system path. Both HTTP and SSH protocols are supported for online Git repos
  (e.g. `[user@]server:project.git`). _Default_: The current project (found by
  walking up from the current working directory tree).

- `stages` - name or names of the stages to retrieve params from. If `None`, all
  parameters from all stages will be retrieved. _Default_: `None`.

- `rev` - Git commit (any [revision](https://git-scm.com/docs/revisions) such as
  a branch or tag name, a commit hash or an
  [experiment](/doc/command-reference/exp) name). If `repo` is not a Git repo,
  this option is ignored. If `None`, current working tree will be used.
  _Default_: `None`

- `deps` - whether to retrieve only params that are stage dependencies or not.
  Accepts `True` or `False` (_default_).

## Example: Filter by stage name(s)

> Working on https://github.com/iterative/example-get-started

`stages` can be a single name (string):

```py
import json
import dvc.api
params = dvc.api.get_params(stages="featurize")
print(json.dumps(params, indent=2))
```

```json
{
  "featurize": {
    "max_features": 200,
    "ngrams": 2

>   }

Reviewer 2: @daavoo in https://github.com/iterative/dvc/pull/7613#issuecomment-1158729997 you said "we don't include the stage name ... when loading a single stage's params" yet this example does show it. Can you confirm what's the behavior? Thanks

Reviewer 2: Ooooh never mind, I read that comment again and I see it's just the structure of the params file.

# dvc.api.params_show()

Returns all the <abbr>params</abbr> associated with the given <abbr>DVC

> repository</abbr>.

Author: No, it won't work

> repository</abbr>.

Author: Not sure about `for the <abbr>stages</abbr>` part, because stages only come into play when using the `stages` argument

> repository</abbr>.

Reviewer 2: > it won't work

Is that expected? `dvc params` CLI works with non-Git DVC projects I believe.

> repository</abbr>.

Author: I misunderstood. I thought you asked about non-dvc projects. 

Non-Git DVC projects do work

> repository</abbr>.

Author: > dvc params CLI works with non-Git DVC projects I believe.

It doesn't because CLI only has `dvc params diff`, which makes sense to don't be supported in non-Git.
`metrics show` does work and `params show` (if/when added to CLI), should work

```py
def params_show(
    *targets: str,
    repo: Optional[str] = None,

>     stages: Optional[Union[str, Iterable[str]]] = None,

Author: > It's still actually optional, right? Python syntax may be confusing here.

Optional is used in type hints to indicate that `None` is also a valid value. From https://docs.python.org/3/library/typing.html#typing.Optional:

> Note that this is not the same concept as an optional argument, which is one that has a default. An optional argument with a default does not require the Optional qualifier on its type annotation just because it is optional.

>     stages: Optional[Union[str, Iterable[str]]] = None,

Reviewer 2: Thanks for the reminder!

> an optional argument, which is one that has a default 

But so `*targets` is also optional, right? Even when it has no default value. Nothing is sent in the Usage block below.

>     stages: Optional[Union[str, Iterable[str]]] = None,

Author: > But so *targets is also optional, right?

It is optional, but the recommended syntax for the `*args` typing is to include only the valid type (in this case, str). `Optional[str]` would mean that `params_show(None)` would be a valid call, which is not.

    rev: Optional[str] = None,
    deps: bool = False,
) -> Dict:
```

#### Usage:

```py
import dvc.api

params = dvc.api.params_show()
```

## Description

> Retrieves the <abbr>params</abbr> tracked in a <abbr>DVC repository</abbr>.

Reviewer 2: What if they're in a params file but not used in any stages. Are they still "tracked" ?

Author: > What if they're in a params file but not used in any stages. Are they still "tracked" ?

They are not tracked and they won't be retrieved unless the untracked params file is passed as `targets`.

Not sure which alternative could be used to better reflect all the posibilities

Reviewer 2: > unless the untracked params file is passed as targets

- [ ] Good to know, we should add this info to the `targets` arg description.

And don't worry, I'll take this text over once all my doubts have been cleared. Thanks

Reviewer 2: No, that's clear enough. I didn't realize it's already mentioned. Thanks!

Without arguments, this function will retrieve all <abbr>params</abbr> from all

> parameter files, for the current revision.

Reviewer 2: What about unstaged changes? Does this read the actual files in the working tree, or the ones in `HEAD`?

Author: > Does this read the actual files in the working tree, or the ones in HEAD?

It reads the actual files in the working tree. Should we replace `current revision` with `working tree` or `workspace`?

Author: I replaced with `working tree`

See the options below to restrict the <abbr>params</abbr> retrieved.

## Parameters

> - **`*targets`** (str, optional): Names of the parameter files to retrieve

Reviewer 2: Please only bold required args, and replicate existing format in other pages. Thanks
```suggestion
- `*targets` - Names of the parameter files to retrieve e.g. "params.py, myparams.toml"
```
☝🏼 Just an idea. Please fix sample value somewhere in the paragraph, it's unclear what format this arg expects.

  <abbr>params</abbr> from. If no `targets` are provided, all parameter files
  will be used. Note that targets don't necessarily have to be defined in
  `dvc.yaml`

- **`repo`** (str, optional): location of the <abbr>DVC repository</abbr>.
  Defaults to the current project (found by walking up from thecurrent working
  directory tree). It can be a URL or a file system path. Both HTTP and SSH
  protocols are supported for online Git repos (e.g. [user@]server:project.git).

- **`stages`** (Union[str, Iterable[str]], optional): Name(s) of the stages to
  retrieve <abbr>params</abbr> from. Defaults to `None`. If no stages are
  provided, all parameters from all stages will be retrieved.

- **`rev`**: (str, optional): Name of the Git
  [revision](https://git-scm.com/docs/revisions) to retrieve <abbr>params</abbr>

>   from. Example of git revision can be a branch or tag name, a commit hash or a

Reviewer 2: Let's reuse some text [from `get_url()`](https://dvc-org-api-get-params-2lnv5kl.herokuapp.com/doc/api-reference/get_url#parameters), I think.

Author: Done

  dvc experiment name. Defaults to `HEAD`. If `repo` is not a Git repo, this
  option is ignored.

- **`deps`**: (bool, optional): Whether to retrieve only <abbr>params</abbr>
  that are <abbr>stage</abbr> dependencies or not. Defaults to `False`.

> ## Example: No arguments.

Reviewer 2: ```suggestion
## Example: default behavior (no filters)
```

> Working on https://github.com/iterative/example-get-started

```py
import json
import dvc.api
params = dvc.api.get_params()
print(json.dumps(params, indent=4))
```

```json
{
  "prepare": {
    "split": 0.2,
    "seed": 20170428
  },
  "featurize": {
    "max_features": 200,
    "ngrams": 2
  },
  "train": {
    "seed": 20170428,
    "n_est": 50,
    "min_split": 0.01
  }
}
```

> ## Example: Filtering with `stages`.

Reviewer 2: ```suggestion
## Example: Filter by `stages`
```

> Working on https://github.com/iterative/example-get-started

`stages` can a single string:

```py
import json
import dvc.api
params = dvc.api.get_params(stages="prepare")
print(json.dumps(params, indent=2))
```

```json
{
  "prepare": {
    "split": 0.2,
    "seed": 20170428
  }
}
```

Or an iterable of strings:

```py
import json
import dvc.api
params = dvc.api.get_params(stages=["prepare", "train"])
print(json.dumps(params, indent=2))
```

```json
{
  "prepare": {
    "split": 0.2,
    "seed": 20170428
  },
  "train": {
    "seed": 20170428,
    "n_est": 50,
    "min_split": 0.01
  }
}
```

> ## Example: Using `rev`.

Reviewer 2: ```suggestion
## Example: Specify a project version
```

> Working on https://github.com/iterative/example-get-started

```py
import json
import dvc.api

> params = dvc.api.get_params(rev="tune-hyperparams")

Reviewer 2: Needs a bit more explanation I think. Unclear what's happening or why do this from just the command and output samples (which look similar to the other examples).

Author: Added an introductory line.

print(json.dumps(params, indent=2))
```

```json
{
  "prepare": {
    "split": 0.2,
    "seed": 20170428
  },
  "featurize": {
    "max_features": 200,
    "ngrams": 2
  },
  "train": {
    "seed": 20170428,
    "n_est": 100,
    "min_split": 8
  }
}
```

> ## Example: Using `targets`.
> 
> > Working on `multi-params-files` folder of
> > https://github.com/iterative/pipeline-conifguration
> 
> You can pass a single target:

Reviewer 2: ```suggestion
## Example: Filter by parameter files
> Working on
> https://github.com/iterative/pipeline-conifguration/tree/main/multi-params-files

You can pass a single param file as target:
```

```json
{
  "run_mode": "prod",
  "configs": {
    "dev": "configs/params_dev.yaml",
    "test": "configs/params_test.yaml",
    "prod": "configs/params_prod.yaml"
  },
  "evaluate": {
    "dataset": "micro",
    "size": 5000,
    "metrics": ["f1", "roc-auc"],
    "metrics_file": "reports/metrics.json",
    "plots_cm": "reports/plot_confusion_matrix.png"
  }
}
```

Or multiple targets:

```py
import json
import dvc.api
params = dvc.api.params_show(
  "configs/params_dev.yaml", "configs/params_prod.yaml")
print(json.dumps(params, indent=2))
```

```json
{
  "configs/params_prod.yaml:run_mode": "prod",
  "configs/params_prod.yaml:config_file": "configs/params_prod.yaml",
  "configs/params_prod.yaml:data_load": {
    "dataset": "large",
    "sampling": {
      "enable": true,
      "size": 50000
    }
  },
  "configs/params_prod.yaml:train": {
    "epochs": 1000
  },
  "configs/params_dev.yaml:run_mode": "dev",
  "configs/params_dev.yaml:config_file": "configs/params_dev.yaml",
  "configs/params_dev.yaml:data_load": {
    "dataset": "development",
    "sampling": {
      "enable": true,
      "size": 1000
    }
  },
  "configs/params_dev.yaml:train": {
    "epochs": 10
  }
}
```

> ## Example: Git URL as `repo`.

Reviewer 2: ```suggestion
## Example: Show parameters from a remote DVC repository
```

```py
import json
import dvc.api
params = dvc.api.get_params(

>     repo="https://github.com/iterative/demo-fashion-mnist")

Reviewer 2: This one also needs some explanation/motivation I think. Or if it's basically the same as the others, we probably don't need the example 🙂 (the motivation can be in the **Parameters** section).

Author: Motivation is that you can use the API to retrieve params from a repository without having to clone and/or navigate to it locally

# dvc.api.params_show()

Load <abbr>parameters</abbr> (name and values) tracked in a <abbr>DVC
project</abbr>.

```py
def params_show(
    *targets: str,  # Optional

>     repo: Optional[str] = None,

Reviewer 2: WDYT about this comment to make it extra clear @daavoo, @dberenbaum ? Thanks

# dvc.api.params_show()

Load <abbr>parameters</abbr> (name and values) tracked in a <abbr>DVC
project</abbr>.

```py
def params_show(
    *targets: str,  # Optional
    repo: Optional[str] = None,
    stages: Optional[Union[str, Iterable[str]]] = None,
    rev: Optional[str] = None,
    deps: bool = False,
) -> Dict:
```

## Usage:

```py
import dvc.api

params = dvc.api.params_show()
```

## Description

Retrieves <abbr>params</abbr> keys and values from a <abbr>DVC project</abbr>
and returns a JSON structure such as:

```json
{
  "split": 0.2,
  "seed": 20170428
}
```

Without arguments, this function will retrieve all params from all tracked

> parameter files (used in any `dvc.yaml` file). This applies to the current

Reviewer 2: Please confirm this is correct @daavoo , thanks.

Author: 👍

Reviewer: I think it will only retrieve params from `dvc.yaml` in the current directory, right? See https://github.com/iterative/dvc/issues/7926.
